catalog/core/catalog-core-migratable/src/main/java/org/codice/ddf/catalog/migratable/impl/MigrationTaskManager.java

```diff
@@ -0,0 +1,194 @@
+        this.taskExecutor = MoreExecutors.listeningDecorator(executor);
+        this.taskFailure = createAtomicReference();
+        this.catalogConfig = config;
+        this.fileWriter = fileWriter;
+    }
+
+    AtomicReference<Throwable> createAtomicReference() {
+        return new AtomicReference<>(null);
+    }
+
+    /**
+     * Must be called prior to exportFinish and exportMetacardQuery to initilize the
+     * export directory.
+     *
+     * @throws MigrationException thrown if the setup fails
+     */
+    public void exportSetup() throws MigrationException {
+        fileWriter.init(this.catalogConfig.getExportPath());
+    }
+
+    /**
+     * Must be called after exportSetup and all calls to exportMetacardQuery. This will attempt
+     * to gracefully shutdown the executor service and handle any exceptions that might have
+     * occured in the asynchronous threads.
+     * <p>
+     * If you are using {@link MigrationTaskManager}; that is, you have constructed an instance,
+     * you must call this method to gracefully release resources and shutdown.
+     *
+     * @throws MigrationException thrown if some of the some of the metacards couldn't be exported
+     */
+    public void exportFinish() throws MigrationException {
+        LOGGER.info("Attempting to shutdown catalog export");
+        taskExecutor.shutdown();
+        boolean isGracefulTermination = false;
+        try {
+            isGracefulTermination = taskExecutor.awaitTermination(1L, TimeUnit.MINUTES);
+        } catch (InterruptedException e) {
+            LOGGER.warn("Executor was interrupted during shutdown: " + e.getMessage(), e);
+        } finally {
+            if (!isGracefulTermination) {
+                taskExecutor.shutdownNow();
+            }
+
+            checkForFailures();
+        }
+    }
+
+    /**
+     * Creates a sublist from the given results and a destination file for metacards based on
+     * the {@link CatalogMigratableConfig} object used by this task manager. An asynchronous
+     * task is started for exporting the metacards and is deferred to the appropriate instance
+     * of {@link MigrationFileWriter} for processing.
+     *
+     * @param results          The results of a catalog query that need to be exported.
+     * @param exportGroupCount The group or page number of the query. This value is not monitored
+     *                         or audited for repeats and is strictly for record keeping by naming
+     *                         the resulting export files appropriately.
+     * @throws MigrationException Thrown if one of the writing threads fails or throws an exception
+     *                            itself.
+     */
+    public void exportMetacardQuery(final List<Result> results, long exportGroupCount)
+            throws MigrationException {
+        for (int i = 0; i < results.size(); i += catalogConfig.getExportCardsPerFile()) {
```

Comment:
We should try to use http://guava-libraries.googlecode.com/svn/tags/release09/javadoc/com/google/common/collect/Lists.html#partition%28java.util.List,%20int%29 here instead.